hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileUpdateHandler.java

```diff
@@ -104,6 +118,21 @@ public void handle(@NonNull final HandleContext handleContext) throws HandleExce
         fileStore.put(builder.build());
     }
 
+    private void handleUpdateUpgradeFile(FileUpdateTransactionBody fileUpdate, HandleContext handleContext) {
+        final var fileStore = handleContext.writableStore(WritableUpgradeStore.class);
+        // empty old upgrade file
+        while (fileStore.poll() != null)
+            ;
+        final var file = new File.Builder()
+                .fileId(fileUpdate.fileIDOrThrow())
+                .contents(fileUpdate.contents())
+                .deleted(false)
+                .expirationTime(fileUpdate.expirationTimeOrElse(EXPIRE_NEVER).seconds())
```

Comment:
Do we really want upgrade files to never expire? Seems like setting them to expire in a week or a month would be more than enough